### PR TITLE
feat(*): Add the ability to claim a cluster by version

### PR DESCRIPTION
**Pull request by jchauncey**
- adds --cluster-version cli flag
- it will attempt 10 times to find a random cluster that matches the version provided

### CODE REVIEW

File: handlers/delete_lease_test.go [hunk 1]
```diff
@@ -85,10 +85,11 @@ func TestDeleteLeaseNoSuchLease(t *testing.T) {
 }
 
 func TestDeleteLeaseExists(t *testing.T) {
-	clusterNames := []string{"cluster1", "cluster2"}
-	uuids := make([]uuid.UUID, len(clusterNames))
+	// clusterNames := []string{"cluster1", "cluster2"}
```
Comment (vdice) [inline handlers/delete_lease_test.go]: should be able to remove this line now

File: handlers/gke_test.go [hunk 2]
```diff
@@ -21,53 +20,59 @@ const (
...
-	for uuidStr := range rawAnnotations {
-		parsedUUID := uuid.Parse(uuidStr)
-		assert.True(t, parsedUUID != nil, "uuid parsed from %s was invalid", uuidStr)
-		freedUUID = parsedUUID
-		break
+	assert.Equal(t, unusedCluster.Name, "getClusterByVersion", "free cluster name")
+	assert.Equal(t, unusedCluster.CurrentNodeVersion, "1.1.1", "free cluster version")
+}
+
+func TestFindRandonUnusedGKECluster(t *testing.T) {
```
Comment (vdice) [inline handlers/gke_test.go]: just a typo here s/Randon/Random

File: testutil/annotations.go [hunk 3]
```diff
@@ -13,22 +14,24 @@ func DefaultTimeFunc(i int) time.Time {
 
 // DefaultUUIDFunc can be passed to GetRawAnnotations for the uuidFunc parameter
 func DefaultUUIDFunc(i int) uuid.UUID {
-	return uuid.NewUUID()
+	return uuid.NewRandom()
+	// return uuid.NewUUID()
```
Comment (vdice) [inline testutil/annotations.go]: looks like this can be removed

**Comment by mboersma** [comment]
This looks good, although the comments made by @vdice should be implemented.

**Comment by jchauncey** [comment]
We found some issues in testing so I need to remove the waiting review
label.

File: handlers/search_free_cluster.go [hunk 4]
```diff
@@ -23,29 +23,32 @@ func (e errExpiredLeaseGKEMissing) Error() string {
...
-	if unusedClusterErr != nil && expiredLeaseErr != nil {
-		return nil, errNoAvailableOrExpiredClustersFound{}
-	}
-	var availableCluster *container.Cluster
-	if unusedCluster != nil {
-		availableCluster = unusedCluster
-	}
+func searchForFreeCluster(clusterMap *clusters.Map, leaseMap *leases.Map, clusterRegex string, clusterVersion string) (*container.Cluster, error) {
+	uuidAndLeases, expiredLeaseErr := findExpiredLeases(leaseMap)
+	// if unusedClusterErr != nil && expiredLeaseErr != nil {
```
Comment (vdice) [inline handlers/search_free_cluster.go]: some leftover commented-out code here

**Comment by vdice** [comment]
@jchauncey other than https://github.com/deis/k8s-claimer/pull/84/files#r76109096 code LGTM and tested manually to success!

**Pull request merged**